src/model_io/urdf-kdl/src/urdf_import.cpp

```diff
@@ -377,6 +377,14 @@ bool framesFromKDLTree(const KDL::Tree& tree,
             std::string parentLinkName = GetTreeElementSegment(GetTreeElementParent(seg->second)->second).getName();
             framesNames.push_back(frameName);
             parentLinkNames.push_back(parentLinkName);
+
+            //also check parent
+            KDL::Segment parent = GetTreeElementSegment(GetTreeElementParent(seg->second)->second);
+            if (parent.getJoint().getType() == KDL::Joint::None &&
+                parent.getInertia().getMass() == 0.0)
+            {
+                framesNames.push_back(parentLinkName);
+            }
```

Comment:
Just for safety, I would add a check that the parent has only one child, otherwise we risk to have the situation in which a "fake link (frame)" is a parent to a "real link". Adding such check should move us closer to the behavior that we get in non-KDL classes by two calls to `removeFakeLinks`, as described in https://github.com/robotology/idyntree/issues/247 .